### PR TITLE
feat: routes sanity checks

### PR DESCRIPTION
**Pull request by giulianok**
Adding some sanity checks for routes to make sure the right headers, response, and status code are in place

## Description

Added a new integration test case to check some of the routes' headers, response, and status code

## Motivation and Context

This makes the integration tests stronger and helps the fastify migration process to make sure everything is in place

## How Has This Been Tested?

Integration tests

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: __tests__/integration/one-app.spec.js [hunk 1]
```diff
@@ -1668,3 +1668,213 @@ describe('heapdump', () => {
...
+    expect(rawHeaders['content-length']).toEqual(['2']);
+    expect(rawHeaders['content-type']).toEqual(['text/plain; charset=utf-8']);
+    expect(rawHeaders['referrer-policy']).toEqual(['same-origin']);
+    expect(rawHeaders['strict-transport-security']).toEqual([
+      'max-age=15552000; includeSubDomains',
+    ]);
+    expect(rawHeaders.vary).toEqual(['Accept-Encoding']);
+    expect(rawHeaders['x-content-type-options']).toEqual(['nosniff']);
+    expect(rawHeaders['x-frame-options']).toEqual(['DENY']);
+    expect(rawHeaders['x-xss-protection']).toEqual(['1; mode=block']);
```
Comment (10xLaCroixDrinker) [inline __tests__/integration/one-app.spec.js:1729]: What about an inline snapshot of the headers?

**Review by 10xLaCroixDrinker — APPROVED** [review]

File: __tests__/integration/one-app.spec.js (hunk 1 above)
Comment (giulianok) [reply __tests__/integration/one-app.spec.js:1729]: we could use inline snapshot, only problem is that some header values are dynamic, like `date` and `etag`, maybe there's a way to mock them

Comment (10xLaCroixDrinker) [reply __tests__/integration/one-app.spec.js:1729]: check out asymmetric snapshots

File: __tests__/integration/one-app.spec.js [hunk 2]
```diff
@@ -1668,3 +1705,323 @@ describe('heapdump', () => {
...
+        expect.any(String),
+      ],
+      etag: [
+        expect.any(String),
+      ],
+      'expect-ct': [
+        'max-age=0',
+      ],
+      'one-app-version': [
+        '6.0.0-0520af2e',
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js:1865]:
```suggestion
        expect.any(String),
```

File: __tests__/integration/one-app.spec.js [hunk 3]
```diff
@@ -1668,3 +1705,323 @@ describe('heapdump', () => {
     expect(size).toBeLessThanOrEqual(100 * oneMegabyte);
   });
 });
+
+describe('Routes sanity checks', () => {
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js:1709]: these should be moved to be in the first block https://github.com/americanexpress/one-app/blob/main/__tests__/integration/one-app.spec.js#L59

Comment (10xLaCroixDrinker) [inline __tests__/integration/one-app.spec.js:1709]:
```suggestion
describe('Routes confidence checks', () => {
```

The term "sanity check" is ableist

File: __tests__/integration/one-app.spec.js [hunk 4]
```diff
@@ -1668,3 +1705,323 @@ describe('heapdump', () => {
...
+  test('Request: /_/report/errors responds with status 415', async () => {
+    const response = await fetch(`${fetchUrl}/_/report/errors`, {
+      ...defaultFetchOptions,
+      method: 'POST',
+      headers: {
+        origin: 'test.example.com',
+      },
+    });
+
+    expect(response.headers.raw()).toEqual({
```
Comment (10xLaCroixDrinker) [inline __tests__/integration/one-app.spec.js:1842]: Do we need all the headers for these tests or can we use `expect.objectContaining` to target the most relevant ones?

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

File: __tests__/integration/one-app.spec.js (hunk 4 above)
Comment (giulianok) [reply __tests__/integration/one-app.spec.js:1842]: we could, I thought the feedback about using inline snapshot (https://github.com/americanexpress/one-app/pull/802#discussion_r938331534) was about checking all the headers beside readability. In my opinion, I think it's good to check all the them, especially if a new one gets added in the future, the `toEqual` assertion will enforce the check of all headers including the new ones

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Comment by 10xLaCroixDrinker** [comment]
the commit type should have been test rather than feat

**Pull request merged**